stacks-signer/src/signerdb.rs

```diff
@@ -420,6 +430,34 @@ mod tests {
             .is_none());
     }
 
+    #[test]
+    fn test_has_pending_blocks() {
```

Comment:
This is kind of a weird test because we're basically creating 4 duplicate blocks, when the test is really only testing that if the 1 block has `valid` set, then there are no pending ones. IMO the right test would be to have 2+ blocks in different reward cycles (using `create_block_override`), updating one of them, then testing `has_pending_blocks`.

Reply:
Good catch. I legit meant to update this test but completely forgot about it. Fixed.